tools/stylelint/no-concrete-rules/index.js

```diff
@@ -0,0 +1,39 @@
+const stylelint = require('stylelint');
+const path = require('path');
+const ruleName = 'material/no-concrete-rules';
+const messages = stylelint.utils.ruleMessages(ruleName, {
+  expected: pattern => `CSS rules must be placed inside a mixin for files matching '${pattern}'.`
+});
+
+/**
+ * Stylelint plugin that will log a warning for all top-level CSS rules.
+ * Can be used in theme files to ensure that everything is inside a mixin.
+ */
+const plugin = stylelint.createPlugin(ruleName, (isEnabled, options) => {
+  return (root, result) => {
+    if (!isEnabled) return;
+
+    const filePattern = new RegExp(options.filePattern);
+    const fileName = path.basename(root.source.input.file);
+
+    if (!filePattern.test(fileName)) return;
+
+    // Go through all the nodes and report a warning for every CSS rule.
+    // We use a regular `forEach`, instead of the PostCSS walker utils,
+    // because we only care about the top-level nodes.
+    root.nodes.forEach(node => {
+      if (node.type === 'rule') {
```

Comment:
One note here @jelbourn: this will capture Sass placeholder rules (e.g. `%something {}`) which won't produce code unless they're extended. I opted to report them as well, because people can do things like `%something, .something-else` which does produce CSS.

Reply:
I'm on board with that. We generally don't use placeholders (I think just a couple in button). I'm a fan of the greater control mixins give you about what's being rendered.